Remove vestigial nodes

rgov opened the pull request:
> [!NOTE]  
> This is part of a general project cleanup along the road to ROS 2.

1. Remove the instant_replay_node. The idea here was that it would buffer a few seconds of frames from the RTSP feed and then if we hit an error, it would write the buffer to the bag file. It hasn't been used in years and there might be better ways to handle video now.

2. Replace [image_transport_node](https://github.com/WHOIGit/PhytO-ARM/blob/main/src/phyto_arm/src/image_transport_node.py) with a standard ROS [image_transport](https://wiki.ros.org/image_transport) republish node.

    The IFCB node publishes `CompressedImage` messages to `/ifcb/roi/image`. This is a violation of the ROS [image_transport](https://wiki.ros.org/image_transport) pattern which would have us publish to `/ifcb/roi/image/compressed` and then we could use a standard `republish` node to decompress it for the Triton node:

        rosrun image_transport republish compressed in:=/ifcb/roi/image raw out:=/ifcb/roi/image

    If we need to beat it into submission we can use topic remapping to get what we want. 

    This wouldn't affect our bag files because we already exclude `/ifcb/roi/.*` in `rosbag.lauch`.

Am I missing anything?

rgov committed:
Remove vestigial instant replay node

rgov committed:
Replace image_transport_node with standard republish node

rgov commented:
Oh hey I had a bug for this! I was so clever back then. Fixes #28

rgov commented:
> Is instant_replay_node broken? While we haven’t used it in a while, it can be very very helpful. I was actually wondering if we had it configured on SeaTrac today after boat hung up on a lobster pot.

I am not sure the last time it was used – perhaps as far back as 2020. Because an error now brings down all nodes, I don't believe it currently works. If we wanted to bring back the functionality, we should look into using gstreamer or similar to properly capture video; this node takes way too much RAM.

rgov committed:
Remove RBR relay code

figuernd commented:
> @figuernd Any qualms about ditching `mock_profiler_node.py`? The profiler node is a "pure" node that doesn't interact with any external entities, so I believe it is sufficient to mock out the CTD and winch.

Profile values are needing for simulating end-to-end arm_ifcb task behavior

figuernd commented:
> I think [rbr_relay.sh](https://github.com/WHOIGit/PhytO-ARM/blob/main/rbr_relay.sh) can go; it shouldn't be at the root of the repository, and [rbr_relay.service](https://github.com/WHOIGit/PhytO-ARM/blob/main/src/rbr_maestro3_ctd/rbr_relay.service) doesn't even use it.
> 
> When the RBR is attached to the same host running ROS, we should use the ctd_comms node for this:
> 
> ```
>         <node name="ctd_comms" pkg="ds_util_nodes" type="bridge_node" />
> ```
> 
> If we're redirecting traffic from some other host then we can use `socat` though it seems like this trick is not specific to the RBR and we might want to put it somewhere else (`scripts/`?)

Right that script is purely for cases where an external host needs to proxy streams to the IFCB, though if using P-A on an RPi aboard ships where this happens becomes more popular then the need to proxy streams to the IFCB becomes less necessary. At any rate we're not really using the script as you say and it should be deleted, the systemd service with socat is sufficient.

figuernd reviewed (changes requested):
I would like to keep the mock profiler node for local testing purposes. We should probably preserve the relay systemd service somewhere as well until we're sure it won't be used in the field anymore.

rgov commented:
I have rebased onto `main` and reverted the mock profiler mode, as requested.

The `rbr_relay.sh` and `rbr_relay.service` are tracked in Git history. I'm happy to restore them if needed in another change but I'd like to get this PR merged in so I can stack some of the ROS2 changes on top.
